src/query_plan/fetch_dependency_graph.rs

```diff
@@ -130,17 +135,68 @@ pub(crate) struct DeferTracking {
 // TODO: Write docstrings
 #[derive(Debug, Clone)]
 pub(crate) struct DeferredInfo {
-    label: NodeStr,
-    path: FetchDependencyGraphPath,
-    sub_selection: NormalizedSelectionSet,
-    deferred: IndexSet<NodeStr>,
-    dependencies: IndexSet<NodeStr>,
+    pub(crate) label: NodeStr,
+    pub(crate) path: FetchDependencyGraphPath,
+    pub(crate) sub_selection: NormalizedSelectionSet,
+    pub(crate) deferred: IndexSet<NodeStr>,
+    pub(crate) dependencies: IndexSet<NodeStr>,
 }
 
 // TODO: Write docstrings
 #[derive(Debug, Clone)]
 pub(crate) struct FetchDependencyGraphPath {
-    full_path: OpPath,
-    path_in_node: OpPath,
-    response_path: Vec<FetchDataPathElement>,
+    pub(crate) full_path: OpPath,
+    pub(crate) path_in_node: OpPath,
+    pub(crate) response_path: Vec<FetchDataPathElement>,
+}
+
+impl FetchDependencyGraphNode {
+    pub(crate) fn cost(&mut self) -> Result<QueryPlanCost, FederationError> {
+        if self.cached_cost.is_none() {
+            self.cached_cost = Some(self.selection_set.selection_set.cost(1)?)
+        }
+        Ok(self.cached_cost.unwrap())
+    }
+
+    // TODO: https://github.com/apollographql/federation/blob/f69a0694b95/query-planner-js/src/buildPlan.ts#L1518-L1573
+    pub(crate) fn to_plan_node(
+        &self,
+        _config: &QueryPlannerConfig,
+        _handled_conditions: &Conditions,
+        _variable_definitions: &[Node<VariableDefinition>],
+        _fragments: Option<&RebasedFragments>,
+        _op_name: Option<String>,
+    ) -> Option<super::PlanNode> {
+        todo!()
+    }
+}
```

Comment:
TODO

Reply:
I’ve filed https://github.com/apollographql/federation-next/issues/158 to track following up on this.